src/category_theory/localization/predicate.lean

```diff
@@ -41,4 +49,132 @@ instance Q_is_localization : W.Q.is_localization W :=
 
 end functor
 
+namespace localization
+
+/-- This universal property states that a functor `L : C ⥤ D` inverts morphisms
+in `W` and the all functors `D ⥤ E` (for a fixed category `E`) uniquely factors
+through `L`. -/
+@[nolint has_nonempty_instance]
+structure strict_universal_property_fixed_target :=
+(inverts : W.is_inverted_by L)
+(lift : Π (F : C ⥤ E) (hF : W.is_inverted_by F), D ⥤ E)
+(fac : Π (F : C ⥤ E) (hF : W.is_inverted_by F), L ⋙ lift F hF = F)
+(uniq : Π (F₁ F₂ : D ⥤ E) (h : L ⋙ F₁ = L ⋙ F₂), F₁ = F₂)
+
+/-- The localized category `W.localization` that was constructed satisfies
+the universal property of the localization. -/
+def strict_universal_property_fixed_target.for_Q :
+  strict_universal_property_fixed_target W.Q W E :=
+{ inverts := W.Q_inverts,
+  lift := construction.lift,
+  fac := construction.fac,
+  uniq := construction.uniq, }
+
+/-- When `W` consists of isomorphisms, the identity satisfies the universal property
+of the localization. -/
+def strict_universal_property_fixed_target.for_id (hW : W ⊆ morphism_property.isomorphisms C):
+  strict_universal_property_fixed_target (𝟭 C) W E :=
+{ inverts := λ X Y f hf, hW f hf,
+  lift := λ F hF, F,
+  fac := λ F hF, by { cases F, refl, },
+  uniq := λ F₁ F₂ eq, by { cases F₁, cases F₂, exact eq, }, }
+
+end localization
+
+namespace functor
+
+variables (h₁ : localization.strict_universal_property_fixed_target L W D)
+  (h₂ : localization.strict_universal_property_fixed_target L W W.localization)
+
+namespace is_localization.mk'
+
+lemma unit_eq :
+  𝟭 W.localization = localization.construction.lift L h₁.inverts ⋙ h₂.lift W.Q W.Q_inverts :=
+begin
+  apply localization.construction.uniq,
+  rw [← functor.assoc, localization.construction.fac, h₂.fac, functor.comp_id],
+end
+
+lemma counit_eq :
+  h₂.lift W.Q W.Q_inverts ⋙ localization.construction.lift L h₁.inverts = 𝟭 D :=
+begin
+  apply h₁.uniq,
+  rw [← functor.assoc, h₂.fac, localization.construction.fac, functor.comp_id],
+end
+
+/-- The equivalence of categories `W.localization ≅ D` obtained when `L : C ⥤ D`
+satisfies the universal property of the localization. -/
+def equivalence : W.localization ≌ D :=
```

Comment:
```suggestion
@[simps]
def equivalence : W.localization ≌ D :=
```

Reply:
Actually, this is a definition that is only used to prove a `Prop` so that we do not really care about the fields of this equivalence. I have inlined the code for this in the proof of `is_localization.mk'`.

When it is needed, a choice of equivalence `W.localization ≌ D` shall be provided by `equivalence_from_model`.